node/src/snark_pool/snark_pool_reducer.rs

```diff
@@ -202,13 +205,17 @@ impl SnarkPoolState {
                     }
                 }
 
-                // TODO: libp2p logic already broadcasts everything right now and doesn't
-                // wait for validation, thad needs to be fixed. See #952
                 dispatcher.push(P2pChannelsSnarkAction::Libp2pBroadcast {
                     snark: snark.clone(),
                     nonce: 0,
                     is_local: *is_sender_local,
                 });
+
+                dispatcher.push(P2pNetworkPubsubAction::BroadcastValidatedMessage {
```

Comment:
This doesn't feel right, how come we have both this and `Libp2pBroadcast` above ? Please revise the flow, because `Libp2pBroadcast` eventually ends dispatching `BroadcastValidatedMessage` too.